File: Lua/problem061.lua
-- You need to initialize a table for exporting the module.
problem061 = {}

function problem061.alt_pow(base, exponent)
    if exponent == 0 then
        return 1
    elseif exponent % 2 == 0 then
        return math.pow(problem061.alt_pow(base, exponent / 2), 2)
    else
        return base * problem061.alt_pow(base, exponent - 1)
    end
end

return problem061

File: JavaScript/problem061.js
const altPow = (base, exponent) => {
  if (exponent === 0) {
    return 1
  } else if (exponent % 2 === 0) {
    return Math.pow(altPow(base, exponent / 2), 2)
  } else {
    return base * altPow(base, exponent - 1)
  }
};

module.exports = { altPow }
